tests/lib/rules/jsx-key.js

```diff
@@ -47,46 +46,117 @@ ruleTester.run('jsx-key', rule, {
     {code: 'var App = () => <div />;'},
     {code: '[1, 2, 3].map(function(x) { return; });'},
     {code: 'foo(() => <div />);'},
-    {code: 'foo(() => <></>);', parser: parsers.BABEL_ESLINT},
-    {code: '<></>;', parser: parsers.BABEL_ESLINT}
+    {
+      code: `
+        <div>
+          <App key="one" />
+          <div key="two" />
+          <App key="one" />
+        </div>
+      `,
+      options: [{checkUniquePropKey: false}]
+    },
+    {
+      code: '[<App key="test" />, <App key="test" />];',
+      options: [{checkUniquePropKey: false}]
+    },
+    `
+      <div>
+        <App key="one" />
+        <App key="two" />
+      </div>
+    `
   ],
-  invalid: [].concat({
-    code: '[<App />];',
-    errors: [{message: 'Missing "key" prop for element in array'}]
-  }, {
-    code: '[<App {...key} />];',
-    errors: [{message: 'Missing "key" prop for element in array'}]
-  }, {
-    code: '[<App key={0}/>, <App />];',
-    errors: [{message: 'Missing "key" prop for element in array'}]
-  }, {
-    code: '[1, 2 ,3].map(function(x) { return <App /> });',
-    errors: [{message: 'Missing "key" prop for element in iterator'}]
-  }, {
-    code: '[1, 2 ,3].map(x => <App />);',
-    errors: [{message: 'Missing "key" prop for element in iterator'}]
-  }, {
-    code: '[1, 2 ,3].map(x => { return <App /> });',
-    errors: [{message: 'Missing "key" prop for element in iterator'}]
-  }, {
```

Comment:
also no need to reformat lines like these; that just clutters the diff.

Reply:
Yeah, sorry about that. The current indentation make it easier for me to comment and uncomment then while testing and I forgot the change them back.